vllm/v1/engine/output_processor.py

```diff
@@ -17,6 +17,43 @@
                                    RequestStateStats)
 
 
+class RequestOutputCollector:
```

Comment:
Shall we name this RequestOutputQueue?

Since in the ReuqestState, we still call it queue.

Reply:
Also shall we add a unittest for this class?

Reply:
nit: Could have docstring to highlight this is to collect outputs per request to reduce some potential confusion, especially it is now a standalone data structure.